### PR TITLE
PS-275: Replace max_binlog_files with binlog_space_limit (8.0)

### PR DESCRIPTION
**Pull request by inikep**
Implementation details:
- binlog_space_limit can be set with --binglog-space-limit (it's 0 that is disabled by default)
- binlog_space_limit is checked during each write to binlog, server start, log rotate, FLUSH LOGS, RESET MASTER

- binlog_space_total is a total size of binary logs (except the active one)
- binlog_space_total is updated during server start, log rotate, FLUSH LOGS, RESET MASTER, PURGE BINARY LOGS [TO/BEFORE]

### CODE REVIEW

File: mysql-test/suite/binlog/t/percona_binlog_space_limit.test [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+INSERT INTO t1 VALUES (6,6);
+# binary.000001 should be deleted now
+source include/show_binary_logs.inc;
+
+FLUSH LOGS;
+FLUSH LOGS;
+# binary.000002 should be deleted now
+source include/show_binary_logs.inc;
+
+--disable_warnings
```
Comment (laurynas-biveinis) [inline mysql-test/suite/binlog/t/percona_binlog_space_limit.test]: DROP TABLE t1;, no warning manipulation

File: mysql-test/suite/sys_vars/t/binlog_space_limit_basic.test [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#
+# only global
+#
+select @@global.binlog_space_limit;
+--error ER_INCORRECT_GLOBAL_LOCAL_VAR
+select @@session.binlog_space_limit;
+show global variables like 'binlog_space_limit';
+show session variables like 'binlog_space_limit';
+--disable_warnings
```
Comment (laurynas-biveinis) [inline mysql-test/suite/sys_vars/t/binlog_space_limit_basic.test]: Why warnings have to be disabled here?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: mysql-test/suite/sys_vars/t/binlog_space_limit_basic.test (hunk 2 above)
Comment (inikep) [reply mysql-test/suite/sys_vars/t/binlog_space_limit_basic.test]: This is based on `relay_log_space_limit_basic.test`. It works fine without `--disable_warnings` and `--enable_warnings`.

Comment (laurynas-biveinis) [reply mysql-test/suite/sys_vars/t/binlog_space_limit_basic.test]: Ack, please remove then

**Comment by inikep** [comment]
Updated:
a) `DROP TABLE t1` and re-recorded results
b) removed `--disable/enable_warnings` from `binlog_space_limit_basic.test`

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**